api/handlers/dataexplorerhandler.py

```diff
@@ -305,7 +305,39 @@ def _parse_request(self, request_type='search'):
         for f in filters:
             if f.get('terms'):
                 for k,v in f['terms'].iteritems():
-                    modified_filters.append({'terms': {k+'.raw': v}})
+                    if "null" in v:
+                        try:
+                            v.remove("null")
+                        except AttributeError:
+                            v = None
+                        null_filter = {
+                            'bool': {
+                                'should': [
+                                    {
+                                        'bool': {
+                                            'must': [
+                                                {
+                                                    'bool': {
+                                                        'must_not': [
+                                                            {
+                                                                'exists': {'field': k}
+                                                            }
+                                                        ]
+                                                    }
+                                                },
+                                                {"exists":{"field":k.split('.')[0]}}
```

Comment:
Ah, I think I might have been confusing with my original comment about removing this line - I was assuming it would only be dynamically removed when there is no dot notation in the key (a top level key would have no dot notation). Wouldn't we still need it to prevent the return of documents that would never have the key anyway?